omod/src/main/java/org/openmrs/module/fhir2/providers/EncounterFhirResourceProvider.java

```diff
@@ -59,4 +62,18 @@ public Bundle findEncountersByPatientIdentifier(@RequiredParam(name = Encounter.
 		return FhirUtils
 		        .convertSearchResultsToBundle(encounterService.findEncountersByPatientIdentifier(identifier.getIdPart()));
 	}
+	
+	@Search
+	@SuppressWarnings("unused")
+	public Bundle searchEncounter(@OperationParam(name = Encounter.SP_DATE) DateRangeParam date,
+	        @OperationParam(name = Encounter.SP_LOCATION) ReferenceParam location,
+	        @OptionalParam(name = Encounter.SP_SUBJECT, chainWhitelist = { "", Patient.SP_IDENTIFIER, Patient.SP_GIVEN,
```

Comment:
You seem to have mapped `Encounter.SP_SUBJECT` twice. Are you sure this is right?